test/functional/cmdLineTests/valuetypeddrtests/flattenedvaluetypeddrtests.xml

```diff
@@ -0,0 +1,139 @@
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $valueAddr$ i to show the int contents of the reference object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $valueAddr$ i</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0x7FFFFFFF (offset = 0) (ValueInt)</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $arrayAddr$ to display flattened array contents">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $arrayAddr$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">!J9IndexableObject 0x</output>
+		<saveoutput regex="no" type="required" saveName="arrayIndex1Addr" splitIndex="1" splitBy="!j9object " showMatch="yes">[0] = !j9object 0x</saveoutput>
+		<saveoutput regex="no" type="required" saveName="arrayIndex2Addr" splitIndex="1" splitBy="!j9object " showMatch="yes">[1] = !j9object 0x</saveoutput>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $arrayIndex1Addr$.i to show value of the variable i">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $arrayIndex1Addr$.i</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0x7FFFFFFF (offset = 0) (ValueInt)</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $arrayIndex2Addr$.i to show the value of the variable i">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $arrayIndex2Addr$.i</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0xB670C61E (offset = 0) (ValueInt)</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !flatobject $arrayIndex1Addr$ to recursively show the contents of the object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!flatobject $arrayIndex1Addr$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0x7FFFFFFF</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !flatobject $arrayIndex2Addr$ to recursively show the contents of the object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!flatobject $arrayIndex2Addr$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0xB670C61E</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
```

Comment:
also add tests for singleton (non array) types

Reply:
The test on line 78 tests just a regular object not within an with the instruction `!j9object <addr> i`. Is that what you meant?

Reply:
something like `<test id="Run !j9object $objectAddr$ to display container array contents">` where $objectAddr$ is a singleton object